compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -429,14 +429,14 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                             return;
                         }
                         let trait_ref = trait_predicate.to_poly_trait_ref();
-
-                        let (post_message, pre_message, type_def) = self
+                        let (self_ty, _) = self.tcx.short_ty_string(trait_ref.skip_binder().self_ty());
```

Comment:
You don't want to discard the file where this is being written to, we need to present that to the user in a note.

Now that we're expanding how many types are being mentioned, we might want to modify the API to be able to reuse a single file for multiple types, something that we haven't needed so far, to avoid having multiple files from a single error, but for now that's ok.

Reply:
is it not worth implanting in this pr? how would i go about it?

Reply:
Change the `_` to see if it is some, and if so add a `note`, nothing more.

Reply:
and is the note just the contents of file?

Reply:
The note with the path to the file (which is what the second element in the tuple has). That way the user can know where to find the full path that corresponds to the trimmed path being displayed.